src/main/java/com/rabbitmq/stream/impl/ConsumersCoordinator.java

```diff
@@ -426,8 +417,11 @@ private SubscriptionTracker(
       this.subscriptionListener = subscriptionListener;
       this.trackingClosingCallback = trackingClosingCallback;
       this.messageHandler = messageHandler;
-      this.initialCredits = initialCredits;
-      this.additionalCredits = additionalCredits;
+      this.clientReference = new AtomicReference<>();
+      this.consumerFlowControlStrategy = consumerFlowControlStrategyBuilder.build(
+              String.format("[stream=%s, subscriptionId=%s]", stream, String.valueOf(subscriptionIdInClient)),
+              credits -> this.clientReference.get().credit(this.subscriptionIdInClient, credits)
```

Comment:
This is a bit fragile as the `client` can be null and I don't see any error handling in the calling code.

Reply:
`clientReference.get()` shouldn't return `null` because `clientReference` is set before calling `ClientSubscriptionsManager.add`, but I added an ERROR log.
I chose a log over an exception so that the connection thread or its dedicated thread is not disturbed, but this can also be an `IllegalStateException` or similar if you think it's best. Printing the identifier should be enough for debug purposes.

Reply:
There's not much we can do at this point, so just logging the error is fine.